src/libcore/convert.rs

```diff
@@ -352,6 +352,12 @@ pub trait Into<T>: Sized {
 /// [`from`]: trait.From.html#tymethod.from
 /// [book]: ../../book/ch09-00-error-handling.html
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_on_unimplemented(
+    on(
+        all(_Self="&str", T="std::string::String"),
+        note="you can coerce a `{T}` into a `{Self}` by writing `&*variable`"
```

Comment:
Is the use of "`variable`" in error messages something we've already done? It seems odd to have that specific default name.

Reply:
We haven't done it like this but we do have some messages that use placeholders when the snippet is not available. Because this is an on_unimplemented message we don't have access to the snippet at all. I'm not too happy with the copy so wording changes are totally welcome.

Reply:
I skimmed through the uses of `rustc_on_unimplemented` to see what some other messages do, and i liked this one message in the Iterator trait:

https://github.com/rust-lang/rust/blob/89573b3c8b629507130b1ec8beeaf550fdc0e046/src/libcore/iter/traits/iterator.rs#L78-L79

To make it totally fit in, i would write something like ``"to coerce a `{T}` into a `{Self}`, use `&*`"``, but IMO your existing one can work as-is by using "foo" instead of "variable".

Reply:
Changed to "to coerce a `{T}` into a `{Self}`, use `&*` as a prefix" as just suggesting `&*` can confuse newcomers as to _where_ it should go.